DEVELOPING_OPAL/validate/src/test/java/org/opalj/fpcf/fixtures/alias/ShapeTest.java

```diff
@@ -0,0 +1,229 @@
+package org.opalj.fpcf.fixtures.alias;
+
+import org.opalj.fpcf.properties.alias.Alias;
+import org.opalj.fpcf.properties.alias.MayAlias;
+import org.opalj.fpcf.properties.alias.MustAlias;
+import org.opalj.fpcf.properties.alias.NoAlias;
+
+public class ShapeTest {
+
+    public static void main(String[] args) {
+        ShapeTest shapeTest = new ShapeTest();
+        Shape a = shapeTest.addRectangle(createCircle(1.0));
+
+        if (a != null) {
+            System.out.println("circle added");
+        }
+
+        Shape b = shapeTest.addRectangle(createRectangle(1.0, 1.0));
+
+        if (b != null) {
+            System.out.println("rectangle added");
+        }
+
+        Shape c = shapeTest.setCurrentRectangle(createSquare(1.0));
+
+        if (c != null) {
+            System.out.println("square set");
+        }
+
+        Shape d = createShapeNullPossible("circle", 2.0);
+
+        if (d != null) {
+            System.out.println("circle created");
+        }
+
+        Shape e = createShapeNullImpossible("circle", 2.0);
+
+        if (e != null) {
+            System.out.println("circle created");
+        }
+
+        Shape nullShape = createNull();
+
+        if (nullShape == null) {
+            System.out.println("created null");
+        }
+
+        shapeTest.getLastOr(new Square(1.0));
+        shapeTest.getLastOrCircle(new Square(1.0));
+    }
+
+    @Alias(mayAlias = {@MayAlias(reason = "param might be added to field", id = "ShapeTest.shapesAddParam")},
```

Comment:
I think you should use an @Repeatable annotation to avoid these strange nested annotations. Have a look at properties/callgraph/DirectCall and properties/callgraph/DirectCalls for an example.